File: scripts/seed-registry.py
"""
Seed the icon registry by fetching all icons from GitHub API and stamping
them as "old" (30 days ago) so they won't show a NEW badge.
Only icons added after this baseline will appear as new.
"""

import json
import os
import urllib.request
from datetime import datetime, timedelta, timezone

GITHUB_API_URL = "https://api.github.com/repos/luisson10/pixelarticons/contents/svg"
ROOT = os.path.join(os.path.dirname(__file__), "..")
REGISTRY_PATH = os.path.join(ROOT, "icon-registry.json")

old_date = (datetime.now(timezone.utc) - timedelta(days=30)).isoformat()

print("Fetching icon list from GitHub API...")
req = urllib.request.Request(
    GITHUB_API_URL,
    headers={"Accept": "application/vnd.github+json", "User-Agent": "seed-registry"},
)
with urllib.request.urlopen(req) as res:
    data = json.loads(res.read().decode())

all_icons = sorted(
    f["name"].replace(".svg", "")
    for f in data
    if f.get("type") == "file" and f["name"].endswith(".svg")
)
print(f"Found {len(all_icons)} icons on GitHub.")

registry = {}
if os.path.exists(REGISTRY_PATH):
    try:
        with open(REGISTRY_PATH, "r") as f:
            registry = json.load(f)
        print(f"Loaded existing registry with {len(registry)} entries.")
    except Exception:
        print("Could not parse existing registry — starting fresh.")

seeded = 0
for name in all_icons:
    if name not in registry:
        registry[name] = old_date
        seeded += 1

with open(REGISTRY_PATH, "w") as f:
    json.dump(registry, f, indent=2)

print(f"Done. Seeded {seeded} new icons as 'old'. Registry saved to icon-registry.json.")
print(f"Total icons in registry: {len(registry)}")

File: scripts/seed-registry.mjs
/**
 * Seed the icon registry.
 *
 * Usage:
 *   node scripts/seed-registry.mjs           → stamps ALL icons as "old" (30 days ago)
 *   node scripts/seed-registry.mjs --new abc def  → stamps specific icons as "today"
 *
 * Run this after a sync to mark newly added icons as new.
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ROOT = path.join(__dirname, '..');
const REGISTRY_PATH = path.join(ROOT, 'icon-registry.json');
const SVG_DIR = path.join(ROOT, 'svg');

const OLD_DATE = new Date(Date.now() - 30 * 24 * 60 * 60 * 1000).toISOString();
const NOW = new Date().toISOString();

// Parse --new flag
const args = process.argv.slice(2);
const newFlagIdx = args.indexOf('--new');
const explicitNew = newFlagIdx !== -1 ? args.slice(newFlagIdx + 1) : [];

// Read existing registry (or start fresh)
let registry = {};
if (fs.existsSync(REGISTRY_PATH)) {
  registry = JSON.parse(fs.readFileSync(REGISTRY_PATH, 'utf-8'));
}

// Read all SVG names
const allIcons = fs
  .readdirSync(SVG_DIR)
  .filter((f) => f.endsWith('.svg'))
  .map((f) => f.replace('.svg', ''));

let added = 0;
let markedNew = 0;

for (const name of allIcons) {
  if (explicitNew.includes(name)) {
    // Force mark as new
    registry[name] = NOW;
    markedNew++;
  } else if (!registry[name]) {
    // Never seen before → treat as old baseline
    registry[name] = OLD_DATE;
    added++;
  }
  // Already in registry → leave untouched
}

fs.writeFileSync(REGISTRY_PATH, JSON.stringify(registry, null, 2), 'utf-8');

console.log(`Registry updated:`);
console.log(`  ${allIcons.length} total icons`);
console.log(`  ${added} newly seeded as old`);
console.log(`  ${markedNew} explicitly marked as new`);
console.log(`  Written to ${REGISTRY_PATH}`);
